src/Parser/Multipart.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace React\Http\Parser;
+
+use React\Http\File;
+use React\Http\Request;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\ThroughStream;
+
+/**
+ * Parse a multipart body
+ *
+ * Original source is from https://gist.github.com/jas-/5c3fdc26fedd11cb9fb5
+ *
+ * @author jason.gerfen@gmail.com
+ * @author stephane.goetz@onigoetz.ch
+ * @license http://www.gnu.org/licenses/gpl.html GPL License 3
+ */
+class Multipart
+{
+    protected $buffer = '';
+
+    /**
+     * @var string
+     */
+    protected $boundary;
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @param string $boundary
+     * @param Request $request
+     */
+    public function __construct(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+        preg_match("/boundary=\"?(.*)\"?$/", $headers['content-type'], $matches);
+
+        $this->boundary = $matches[1];
```

Comment:
Content-Type of `multipart/form-data` does not have a boundary. The body looks like

```
------WebKitFormBoundaryPr93i9I5JFCTMovV
Content-Disposition: form-data; name="file"; filename="Screen Shot 2016-01-14 at 21.53.53.png"
Content-Type: image/png

�PNG
```

This leads basically to a notice: `Notice: Undefined offset: 1`